wordpress-mobile/react-native-aztec
# Improve compatibility with cocoapods.

This PR fixes some work in progress from #69 , fixes the example project and reconnects the MediaProviders (no more crashes expected).

**The basic concepts:**

- This module depends on Aztec.
- Trying to add this module to the WPiOS project, we notice that Aztec is also a dependency of WPiOS and it's managed via Cocoapods.

**The path:**

We can share Aztec between WPiOS and RNAztec easily if we also import RNAztec using cocoapods in the WPiOS project.

For that reason the `RNTAztecView.podspec` file was created and added to this project, declaring the dependency on Aztec.

**The problem:**

An issue that I encountered doing this, is that importing `RNTAztecView-Swift.h` to have access to Swift declarations from Obj-C won't work when this module is imported as a Framework using Cocoapods (error: file not found). 

**The solution:**

To solve that issue, we have to avoid calling Swift code directly from Obj-C code.

The way that this work on Native Modules is to declare just the interface that will be exposed to JS side in ObjC, without implementing the `@implementation` part, but to do it directly on a `.swift` file.

We can declare those methods using the `RCT_EXTERN_METHOD` macro instead of `RCT_EXPORT_METHOD`.

**To test:**
- Build and run the example project.
- Check that it runs and the images are loaded.
- Checkout the [`gutenberg-mobile` PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/230).
- Check that the events are passed and handled correctly (bold, italic etc...)
